code/TaxonomyTerm.php

```diff
@@ -125,6 +127,29 @@ public function onBeforeDelete()
             $term->delete();
         }
     }
+    /**
+     *  Set the URL segment allowing filtering by url slug
+     *
+     *  {@inheritDoc}
+     */
+    public function onBeforeWrite() {
+
+        parent::onBeforeWrite();
+
+        // Write the URLSegment to allow for Taxonomy navigation only if one is not supplied
+        if ($this->Name && $this->URLSegment == null)
+        {
+            $this->URLSegment = SiteTree::create()->generateURLSegment($this->Name);
```

Comment:
`SiteTree::singleton()` (not sure if SS4 only) or `singleton('SiteTree')` would be better than `SiteTree::create()` here

Reply:
Agreed I'll change this.